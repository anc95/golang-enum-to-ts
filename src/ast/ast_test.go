package ast

import (
	"os"
	"path"
	"testing"

	"github.com/anc95/golang-enum-to-ts/src/token"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestNormal(t *testing.T) {
	wd, _ := os.Getwd()

	a := path.Join(wd, "../test-cases/normal.go")
	parser := token.NewParser(a)
	tokens := parser.Parse()
	ast := AstGenerator{Tokens: tokens, index: -1}

	result := ast.Gen()

	snaps.MatchSnapshot(t, result)
}
