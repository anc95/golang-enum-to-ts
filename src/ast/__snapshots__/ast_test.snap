
[TestNormal - 1]
ast.File{
    Name: "some",
    Body: {
        ast.TypeDeclaration{Id:"Status", Kind:"string"},
        ast.ConstDeclaration{
            Declarators: {
                {Kind:"Status", Id:"Todo", Value:"iota"},
                {Kind:"", Id:"Done", Value:""},
                {Kind:"", Id:"InProgress", Value:""},
            },
        },
        ast.TypeDeclaration{Id:"Sex", Kind:"int"},
        ast.ConstDeclaration{
            Declarators: {
                {Kind:"Sex", Id:"Female", Value:"female"},
                {Kind:"Sex", Id:"Male", Value:"male"},
            },
        },
    },
}
---
